File: lib/features/match/domain/resolution.dart
enum RoundOutcome { playerAWins, playerBWins, draw }

class Resolution {
  Resolution._();

  /// Resolves a round given both players' moves.
  /// Valid moves: 'rock', 'paper', 'scissors'.
  static RoundOutcome resolve(String moveA, String moveB) {
    if (moveA == moveB) return RoundOutcome.draw;

    const beats = {
      'rock': 'scissors',
      'paper': 'rock',
      'scissors': 'paper',
    };

    if (beats[moveA] == moveB) {
      return RoundOutcome.playerAWins;
    }
    return RoundOutcome.playerBWins;
  }
}

File: backend/lib/resolution.js
/**
 * Server-side round resolution (T97).
 *
 * Mirrors the Flutter client's Resolution.resolve():
 *   - ROCK beats SCISSORS
 *   - PAPER beats ROCK
 *   - SCISSORS beats PAPER
 *   - Same move = draw
 *
 * @returns {'player_a_wins' | 'player_b_wins' | 'draw'}
 */
function resolveRound(moveA, moveB) {
  if (moveA === moveB) return 'draw';

  const beats = {
    rock: 'scissors',
    paper: 'rock',
    scissors: 'paper',
  };

  if (beats[moveA] === moveB) return 'player_a_wins';
  return 'player_b_wins';
}

module.exports = { resolveRound };
